test/e2e/syncer/multishard/multishard_test.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multishard
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	kcpkubernetesclientset "github.com/kcp-dev/client-go/kubernetes"
+	"github.com/kcp-dev/logicalcluster/v3"
+	"github.com/stretchr/testify/require"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	appsv1apply "k8s.io/client-go/applyconfigurations/apps/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/kcp-dev/kcp/test/e2e/framework"
+	"github.com/kcp-dev/kcp/test/e2e/syncer/multishard/workspace1"
+	"github.com/kcp-dev/kcp/test/e2e/syncer/multishard/workspace2"
+)
+
+// TestSyncerTunnelFilter ensures that the syncer tunnel will reject trying to access a Pod that is crafted and not actually upsynced.
+func TestSyncingFromMultipleShards(t *testing.T) {
+	t.Parallel()
+	framework.Suite(t, "transparent-multi-cluster")
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+
+	upstreamServer := framework.SharedKcpServer(t)
+	upstreamConfig := upstreamServer.BaseConfig(t)
+
+	shardNames := upstreamServer.ShardNames()
+	if len(shardNames) < 2 {
+		t.Skip("Test requires at least 2 shards")
+	}
+
+	orgPath, _ := framework.NewOrganizationFixture(t, upstreamServer, framework.TODO_WithoutMultiShardSupport())
+	locationPath, locationWs := framework.NewWorkspaceFixture(t, upstreamServer, orgPath, framework.TODO_WithoutMultiShardSupport())
```

Comment:
Curious if we could let these go to any shard instead of the TODO?

Reply:
Well, there are many such TODO throughout the TMC code, and it seems to me that removing them more globally could be done in a distinct PR.
The goal of the test (check we can sync from several workspaces on distinct shards) is met it seems to me.
Do you think we can postpone the global removal of those TODO to a distinct dedicated PR ?

Reply:
Yes